src/main/common/filter.c

```diff
@@ -208,6 +208,10 @@ void laggedMovingAverageInit(laggedMovingAverage_t *filter, uint16_t windowSize,
     filter->movingWindowIndex = 0;
     filter->windowSize = windowSize;
     filter->buf = buf;
+    filter->movingSum = 0;
+    for (uint16_t i = 0; i < windowSize; i++) {
```

Comment:
This can be done easier with `memset(filter->buf, 0, windowSize * sizeof(float))`.

Also, note that it is more efficient to use `unsigned` for things like loop indices, as it stops the compiler from inserting range checks.

Reply:
I'll change it, thanks for the tips! My non-c background is showing, didn't even think of doing it that way.

I chose uint16_t because that's the type the windowSize has, but good to know about the range checks anyways!